torch/testing/_internal/common_nn.py

```diff
@@ -513,7 +513,7 @@ def kldivloss_with_target_no_reduce_test():
 
 
 def kldivloss_no_reduce_test():
-    t = torch.randn(10, 10)
+    t = torch.rand(10, 10)
```

Comment:
The target of `KLDivLoss` can either be a probability `[0, 1]` or a log-probability `(-inf, 0]`. I've fixed the sample inputs here, but there are some more failures due to 

https://github.com/pytorch/pytorch/blob/facff2ec6562d61eab2ccd9f1c16872e54eec92d/torch/testing/_internal/common_nn.py#L5709

in tests that are automatically generated. Not sure what to do about them.

Reply:
The failures are semi-legit:

- they actually show a divergence between the forward and backward pass:
- the divergence is caused by a really odd choice in the forward pass https://github.com/pytorch/pytorch/blob/6a990535154116960767ad422e724250ecfb8325/aten/src/ATen/native/Loss.cpp#L114-L117
  This silently ignores negative inputs in case of a non-log target. Not only is this not documented, but also indicates a bug in the user pipeline. 

Although this would be BC breaking, IMO we should remove this special case for negative values. Of course we still need to keep the continuous extension for `0 * log(0) := 0` in the forward pass.

Reply:
So, the KL divergence is just defined for for probabilities in [0,1]. Even more, there's nothing stopping you from defining the KL-divergence for general measures, so you could think that the inputs are in [0, infty]. Now, enforcing this would mean checking every input which requires some extra computations.

In my opinion, I think it is good to not check that the inputs are positive, and if they are not, you'll end up returning a `-inf`. If this is well-documented and tested, I think it's better than an arbitrary choice of `0` that's not documented.

Now, this is BC-breaking, but it's BC-breaking with respect to an undocumented "feature", so I believe we should be free to fix these things. cc @jbschlosser